File: XBOOK/XBOOK.Web/ClientApp/src/app/pages/_shared/models/tax-invoice/tax-invoice-view.model.ts
import { TaxInvoiceDetail } from './sale-invoice-detail.model';
import { PaymentView } from '../invoice/payment-view.model';
import { ClientView } from '../client/client-view.model';

export class TaxInvoiceView {
    taxInvoiceId: number;
    invoiceSerial: string;
    invoiceNumber: string;
    issueDate: Date;
    dueDate: Date;
    clientID: number;
    clientName: string;
    address: string;
    taxCode: string;
    contactName: string;
    email: string;
    reference: string;
    subTotal: number;
    discRate: number;
    discount: number;
    vatTax: number;
    amountPaid: number;
    note: string;
    term: string;
    status: string;
    bankAccount: string;
    saleInvoiceId: number;
    saleInvDetailView: TaxInvoiceDetail[];
    clientData: ClientView[];
}

File: XBOOK/XBOOK.Data/ViewModels/TaxInvoiceViewModel.cs
using System;
using System.Collections.Generic;

namespace XBOOK.Data.ViewModels
{
    public class TaxInvoiceViewModel
    {
        public long invoiceID { get; set; }
        public string invoiceSerial { get; set; }
        public string invoiceNumber { get; set; }
        public Nullable<System.DateTime> issueDate { get; set; }
        public Nullable<System.DateTime> dueDate { get; set; }
        public Nullable<int> clientID { get; set; }
        public string reference { get; set; }
        public Nullable<decimal> subTotal { get; set; }
        public Nullable<decimal> discRate { get; set; }
        public Nullable<decimal> discount { get; set; }
        public Nullable<decimal> vatTax { get; set; }
        public Nullable<decimal> amountPaid { get; set; }
        public string note { get; set; }
        public string term { get; set; }
        public string status { get; set; }
        public virtual List<TaxInvDetailViewModel> TaxInvDetailView { get; set; }
    }
}
